src/rviz/tool.cpp

```diff
@@ -73,6 +73,7 @@ void Tool::setName(const QString& name)
 {
   name_ = name;
```

Comment:
Please add a check to verify that name will actually change. If not, you can return early.